shared/tests/utility/support/TestYamlConversion.cpp

```diff
@@ -0,0 +1,201 @@
+                    REQUIRE(testValues["vector2"][i][0].as<double>() == epsilon_x);
+                    REQUIRE(testValues["vector2"][i][1].as<double>() == epsilon_y);
+                }
+            }
+        }
+
+        // Vector3d
+        AND_WHEN("A vector is declared using Vector3d values from the YAML node") {
+            std::vector<Eigen::Vector3d> eigenV3dVector = resolve_expression<Eigen::Vector3d>(testValues["vector3"]);
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(testValues["vector3"].size() == eigenV3dVector.size());
+            }
+            AND_THEN("Values at each index should be equivalent") {
+                for (size_t i = 0; i < testValues["vector3"].size(); i++) {
+                    Approx epsilon_x = Approx(eigenV3dVector.at(i).x()).epsilon(ERROR_THRESHOLD);
+                    Approx epsilon_y = Approx(eigenV3dVector.at(i).y()).epsilon(ERROR_THRESHOLD);
+                    Approx epsilon_z = Approx(eigenV3dVector.at(i).z()).epsilon(ERROR_THRESHOLD);
+                    // NOTE: Must test individual xyz values
+                    REQUIRE(testValues["vector3"][i][0].as<double>() == epsilon_x);
+                    REQUIRE(testValues["vector3"][i][1].as<double>() == epsilon_y);
+                    REQUIRE(testValues["vector3"][i][2].as<double>() == epsilon_z);
+                }
+            }
+        }
+
+        // Array conversions
+        // ints
+        AND_WHEN("An array is declared using integers from the YAML node") {
+            std::array<int, 20> intArray = resolve_expression<int, 20>(testValues["ints"]);
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(testValues["ints"].size() == intArray.size());
+            }
+            AND_THEN("Values at each index should be equivalent") {
+                // Check values at each index
+                for (size_t i = 0; i < testValues["ints"].size(); i++) {
+                    REQUIRE(testValues["ints"][i].as<int>() == intArray.at(i));
+                }
+            }
+        }
+        // floats
+        AND_WHEN("An array is declared using floats from the YAML node") {
+            std::array<float, 20> floatArray = resolve_expression<float, 20>(testValues["floats"]);
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(testValues["floats"].size() == floatArray.size());
+            }
+            AND_THEN("Values at each index should be equivalent") {
+                // Check values at each index
+                for (size_t i = 0; i < testValues["floats"].size(); i++) {
+                    REQUIRE(testValues["floats"][i].as<float>() == floatArray.at(i));
+                }
+            }
+        }
+        // doubles
+        AND_WHEN("An array is declared using doubles from the YAML node") {
+            std::array<double, 20> doubleArray = resolve_expression<double, 20>(testValues["doubles"]);
```

Comment:
I'll also stop suggesting `snake_case` too :p 
```suggestion
            std::array<double, 20> double_array = resolve_expression<double, 20>(test_values["doubles"]);
```